OpenRA.Mods.Common/Activities/Air/Fly.cs

```diff
@@ -60,22 +60,34 @@ public Fly(Actor self, Target t, WPos? initialTargetPosition = null, Color? targ
 			this.minRange = minRange;
 		}
 
-		public static void FlyTick(Actor self, Aircraft aircraft, WAngle desiredFacing, WDist desiredAltitude, WVec moveOverride, bool idleTurn = false)
+		public static void FlyTick(Actor self, Aircraft aircraft, WAngle desiredFacing, WDist desiredAltitude, WVec? moveOverride = null,
+			bool idleTurn = false, WAngle? desiredBodyFacing = null)
 		{
 			var dat = self.World.Map.DistanceAboveTerrain(aircraft.CenterPosition);
-			var move = aircraft.Info.CanSlide ? aircraft.FlyStep(desiredFacing) : aircraft.FlyStep(aircraft.Facing);
-			if (moveOverride != WVec.Zero)
-				move = moveOverride;
 
-			var oldFacing = aircraft.Facing;
-			var turnSpeed = idleTurn ? aircraft.IdleTurnSpeed ?? aircraft.TurnSpeed : aircraft.TurnSpeed;
-			aircraft.Facing = Util.TickFacing(aircraft.Facing, desiredFacing, turnSpeed);
+			var speed = aircraft.MovementSpeed;
+			if (idleTurn && aircraft.Info.IdleSpeed >= 0)
+				speed = aircraft.Info.IdleSpeed;
+
+			var move = aircraft.FlyStep(speed, aircraft.FlightFacing);
+			if (moveOverride.HasValue)
+				move = moveOverride.Value;
+
+			var flightTurnSpeed = aircraft.Info.CanSlide ? aircraft.FlightTurnSpeed ?? aircraft.TurnSpeed : aircraft.TurnSpeed;
```

Comment:
Would you be able to add some comments here to document the expected relationship between the different turn speeds, the split facings, and the resulting behaviour? The `if (idleTurn)` can below is particularly hairy.

Reply:
The chained null-coalescing operators here are getting quite ridiculous. Things would be a lot simpler if we interpret the normal `TurnSpeed` as a flight turnspeed and add a `BodyTurnSpeed` instead. That would necessitate changing the `TurnSpeed` into `BodyTurnSpeed` on all helicopters in existing mods though.

Reply:
What if we defaulted `BodyTurnSpeed` to `TurnSpeed` if its not defined?

Reply:
Making `BodyTurnSpeed` default to `TurnSpeed` would be the idea then. But Those helicopters would then still need their `TurnSpeed` to be 255 while their `BodyTurnSpeed` should be whatever their `TurnSpeed` is currently if we want them to keep their instant directional changing behaviour.

Reply:
Ah, I see. This still sounds like it would be a good approach, even with the slight complication of the update rule. IMO it is more intuitive for `TurnSpeed` to apply to the movement and `BodyTurnSpeed` the visuals of the body.